modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/utils/fields.es.js

```diff
@@ -19,15 +19,15 @@ export function checkValidFieldNameCharacter(character) {
 }
 
 export function normalizeFieldName(fieldName) {
-	let nextUpperCase = false;
+	let nextUpperCase = true;
 	let normalizedFieldName = '';
 
 	fieldName = fieldName.trim();
 
 	for (let i = 0; i < fieldName.length; i++) {
 		let item = fieldName[i];
 
-		if (item === ' ') {
+		if (item === ' ' || item === '_') {
```

Comment:
Hi @daiazius, I don't think we should ignore the underlines. This will change our expected behavior. 

Could you explain why you needed this change?

Reply:
I suggest removing the `underscore` in the `getDefaultFieldName` function, which I believe is not used by the Field Reference